utils/hwloc/hwloc-annotate.c

```diff
@@ -599,24 +603,24 @@ int main(int argc, char *argv[])
 			exit(EXIT_FAILURE);
 		}
 		sizevalue = strtoull(argv[1], &end, 0);
-                if (end) {
-                  if (!strcasecmp(end, "kB"))
-                    sizevalue *= 1024ULL;
-                  else if (!strcasecmp(end, "kiB"))
-                    sizevalue <<= 10;
-                  else if (!strcasecmp(end, "MB"))
-                    sizevalue *= 1024ULL*1024ULL;
-                  else if (!strcasecmp(end, "MiB"))
-                    sizevalue <<= 20;
-                  else if (!strcasecmp(end, "GB"))
-                    sizevalue *= 1024ULL*1024ULL*1024ULL;
-                  else if (!strcasecmp(end, "GiB"))
-                    sizevalue <<= 30;
-                  else if (!strcasecmp(end, "TB"))
-                    sizevalue *= 1024ULL*1024ULL*1024ULL*1024ULL;
-                  else if (!strcasecmp(end, "TiB"))
-                    sizevalue <<= 40;
-                }
+		if (end) {
+			if (!strcasecmp(end, "kB"))
+				sizevalue *= 1024ULL;
+			else if (!strcasecmp(end, "kiB"))
+				sizevalue <<= 10;
+			else if (!strcasecmp(end, "MB"))
+				sizevalue *= 1024ULL*1024ULL;
+			else if (!strcasecmp(end, "MiB"))
+				sizevalue <<= 20;
+			else if (!strcasecmp(end, "GB"))
+				sizevalue *= 1024ULL*1024ULL*1024ULL;
+			else if (!strcasecmp(end, "GiB"))
+				sizevalue <<= 30;
+			else if (!strcasecmp(end, "TB"))
+				sizevalue *= 1024ULL*1024ULL*1024ULL*1024ULL;
+			else if (!strcasecmp(end, "TiB"))
+				sizevalue <<= 40;
+		}
```

Comment:
It seems that the treatment is the same whether the extension is `[kMGT]B` or `[kMGT]iB`. Isn't shifting left 10 times and multiplying by `1024` the same ?

Reply:
Looks like commit 4d9a9843a5616fc702575c4701a8c3cba044757f failed to fix [kMGT]B contrary to what the commit message said :)